File: src/main/java/no/nav/fo/veilarbregistrering/db/UtdanningUtils.java
package no.nav.fo.veilarbregistrering.db;

import no.nav.fo.veilarbregistrering.registrering.bruker.besvarelse.UtdanningSvar;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class UtdanningUtils {
    private static Map<UtdanningSvar, String> nuskodeMap;

    static {
        Map<UtdanningSvar, String> nuskodeMapTmp = new HashMap<>();
        nuskodeMapTmp.put(UtdanningSvar.INGEN_UTDANNING, "0");
        nuskodeMapTmp.put(UtdanningSvar.GRUNNSKOLE, "2");
        nuskodeMapTmp.put(UtdanningSvar.VIDEREGAENDE_GRUNNUTDANNING, "3");
        nuskodeMapTmp.put(UtdanningSvar.VIDEREGAENDE_FAGBREV_SVENNEBREV, "4");
        nuskodeMapTmp.put(UtdanningSvar.HOYERE_UTDANNING_1_TIL_4, "6");
        nuskodeMapTmp.put(UtdanningSvar.HOYERE_UTDANNING_5_ELLER_MER, "7");
        nuskodeMapTmp.put(UtdanningSvar.INGEN_SVAR, "9");

        nuskodeMap = Collections.unmodifiableMap(nuskodeMapTmp);
    }

    public static String mapTilNuskode(UtdanningSvar utdanning) {
        return nuskodeMap.get(utdanning);
    }

    public static UtdanningSvar mapTilUtdanning(String nuskode) {
        return nuskodeMap.entrySet().stream()
                .filter(e -> e.getValue().equals(nuskode))
                .map(Map.Entry::getKey)
                .findFirst()
                .orElse(null);
    }
}

File: src/main/java/no/nav/fo/veilarbregistrering/db/registrering/UtdanningUtils.kt
package no.nav.fo.veilarbregistrering.db.registrering

import no.nav.fo.veilarbregistrering.besvarelse.UtdanningSvar

internal object UtdanningUtils {

    private val nuskodeMap: Map<UtdanningSvar, String> = mapOf(
        UtdanningSvar.INGEN_UTDANNING to "0",
        UtdanningSvar.GRUNNSKOLE to "2",
        UtdanningSvar.VIDEREGAENDE_GRUNNUTDANNING to "3",
        UtdanningSvar.VIDEREGAENDE_FAGBREV_SVENNEBREV to "4",
        UtdanningSvar.HOYERE_UTDANNING_1_TIL_4 to "6",
        UtdanningSvar.HOYERE_UTDANNING_5_ELLER_MER to "7",
        UtdanningSvar.INGEN_SVAR to "9",
    )

    fun mapTilNuskode(utdanning: UtdanningSvar): String? = nuskodeMap[utdanning]

    fun mapTilUtdanning(nuskode: String): UtdanningSvar? =
        nuskodeMap.filterValues { it == nuskode }.keys.firstOrNull()
}